pdata/ptrace/json.go

```diff
@@ -113,7 +113,7 @@ func (dest Span) unmarshalJsoniter(iter *jsoniter.Iterator) {
 				iter.ReportError("readSpan.parentSpanId", fmt.Sprintf("parse parent_span_id:%v", err))
 			}
 		case "flags":
-			dest.orig.Flags = iter.ReadUint32()
+			dest.orig.Flags = json.ReadUint32(iter)
```

Comment:
Can we add a test for this?

Reply:
We have unit-tests for the `json.ReadUint32`. Or for what?

Reply:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/internal/json/number_test.go#L67

Reply:
OK that's good